src/main/java/com/tabnine/inline/listeners/InlineCaretListener.kt

```diff
@@ -11,18 +11,25 @@ import com.tabnine.inline.InlineCompletionCache
 class InlineCaretListener(private val completionPreview: CompletionPreview) : CaretListener, Disposable {
     init {
         Disposer.register(completionPreview, this)
-        ApplicationManager.getApplication().invokeLater { completionPreview.editor.caretModel.addCaretListener(this) }
+        completionPreview.editor.caretModel.addCaretListener(this)
     }
 
     override fun caretPositionChanged(event: CaretEvent) {
         if (ApplicationManager.getApplication().isUnitTestMode) {
             return
         }
+        if (isSingleOffsetChange(event)) {
+            return
+        }
 
         Disposer.dispose(completionPreview)
         InlineCompletionCache.instance.clear(event.editor)
     }
 
+    private fun isSingleOffsetChange(event: CaretEvent): Boolean {
+        return event.oldPosition.column + 1 == event.newPosition.column
```

Comment:
add row check as well, because you can move from (x, y) to (x+1,y+1) and you'll mistakenly return true in this case